packages/modal/addon/components/nucleus-modal/dialog.js

```diff
@@ -180,6 +186,56 @@ class Dialog extends Component {
     }
   }
 
+  /**
+  * createFocus
+  *
+  * @method createFocus
+  * @private
+  *
+  */
+  _createFocus() {
+    let modalDialog = this.element;
+    let focusEl = modalDialog && modalDialog.querySelector("[autofocus]");
+    if(focusEl) {
+      focusEl.focus();
+    }
+    else {
+      modalDialog.focus();
+    }
+  }
+
+  /**
+  * loopFocus
+  *
+  * @method loopFocus
+  * @private
+  * @param {any} event
+  */
+  loopFocus(event) {
+    let modalDialog = this.element.querySelector(".nucleus-modal__dialog");
```

Comment:
Can we move this *after* the isTab check? One less querySelection to be performed during every key stroke.

Reply:
Yes. Will do.